### PR TITLE
feat: implemented keysExist and keyExists endpoints

### PR DESCRIPTION
**Pull request by anitarua**
Addresses #473 by exposing the keyExists and keysExist endpoints to match the .NET implementation.

### CODE REVIEW

**Comment by rishtigupta** [comment]
Great work, thanks @anitarua

The PR lgtm, but will defer to @malandis and @bruuuuuuuce judgement on this since haven't added support for this api before in any sdk!

File: packages/core/src/internal/clients/cache/AbstractCacheClient.ts [hunk 1]
```diff
@@ -1228,6 +1230,38 @@ export abstract class AbstractCacheClient implements ICacheClient {
...
+   * @param {string} cacheName - The cache to look in.
+   * @param {string[] | Uint8Array[]} keys - The keys to look up.
+   * @returns {Promise<CacheKeysExist.Response>}
+   * {@link CacheKeysExist.Success} if at least one key is found? if all keys checked?
+   * {@link CacheKeysExist.Error} on failure.
+   */
+  public async keysExist(
+    cacheName: string,
+    keys: string[] | Uint8Array[]
+  ): Promise<CacheKeyExists.Response> {
```
Comment (malandis) [inline packages/core/src/internal/clients/cache/AbstractCacheClient.ts:1260]:
```suggestion
  ): Promise<CacheKeysExists.Response> {
```

**Review by malandis — COMMENTED** [review]
One small change. Looks good otherwise!

File: packages/client-sdk-nodejs/src/internal/data-client.ts [hunk 2]
```diff
@@ -2869,6 +2871,100 @@ export class DataClient implements IDataClient {
...
+    } catch (err) {
+      return new CacheKeysExist.Error(normalizeSdkError(err as Error));
+    }
+
+    this.logger.trace("Issuing 'keysExist' request");
+
+    const result = await this.sendKeysExist(cacheName, this.convertArray(keys));
+
+    this.logger.trace(
+      "'keyExists' request result: %s",
```
Comment (bruuuuuuuce) [inline packages/client-sdk-nodejs/src/internal/data-client.ts:2936]:
```suggestion
      "'keysExists' request result: %s",
```

File: packages/client-sdk-web/src/internal/data-client.ts [hunk 3]
```diff
@@ -2796,6 +2799,101 @@ export class DataClient<
...
+
+    this.logger.trace("Issuing 'keysExist' request");
+
+    const result = await this.sendKeysExist(
+      cacheName,
+      this.convertArrayToB64Strings(keys)
+    );
+
+    this.logger.trace(
+      "'keyExists' request result: %s",
```
Comment (bruuuuuuuce) [inline packages/client-sdk-web/src/internal/data-client.ts:2866]:
```suggestion
      "'keysExist' request result: %s",
```

**Review by malandis — APPROVED** [review]
🎉

**Pull request merged**